recipe/migrations/arm_variant_type.yaml

```diff
@@ -0,0 +1,70 @@
+migrator_ts: 9738229387
+__migrator:
+  use_local: true
+  operation: key_add
+  migration_number:
+    1
+  build_number:
+    1
+  paused: true
+  override_cbc_keys:
+    - cuda_compiler_stub
+  check_solvable: false
+  primary_key: cuda_compiler_version
+  ordering:
+    arm_variant_type:
+      - None
+      - sbsa
+      - tegra
+  additional_zip_keys:  # [linux and aarch64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+    - arm_variant_type  # [linux and aarch64 and os.environ.get("CF_CUDA_ENABLED", "False") == "True"]
+  wait_for_migrators:
+    - cuda129
+    - aarch64 and ppc64le addition
+  commit_message: |2
+    Build for NVIDIA Tegra devices and CUDA 12.9
+
+    This migration adds `arm_variant_type=tegra` to the build matrix to support NVIDIA Tegra
+    devices compatible with CUDA 12.9. Tegra devices compatible with CUDA 13.0 are SBSA
+    compliant, and do not need a separate build. Only Orin (sm_87) and later devices are
+    supported because earlier devices are not supported by CUDA 12.9.
+
+    The `arm-variant` package must also be added to the build requirements of the recipe in
+    order to constrain the CUDA compiler to the correct variant, and the non CUDA 12.9 tegra
+    variants must be skipped.
+
+    ```yaml
+    build:
+      skip: true  # [(arm_variant_type == "tegra") and not (cuda_compiler_version or "").startswith("12")]
+
+    requirements:
+      build:
+        - {{ compiler('cuda') }}
+        - arm-variant * {{ arm_variant_type }}  # [linux and aarch64 and cuda_compiler_version != "None"]
+    ```
+
+    This migrator is not compatible with feedstocks still targeting CUDA 11.
```

Comment:
This is outdated as well now. It's perfectly fine to combine all the cuda migrators, see [here](https://github.com/h-vetinari/cupy-feedstock/tree/tegra) for an example.

The one thing you need to decide is the canonical order between `cuda118` and `arm_variant_type`. Since both are adding `additional_zip_keys:`, the one that comes after must set values for any previously added zip keys as well.

My preference would be to sort `arm_variant_type` _before_ `cuda118` (which is what I did in https://github.com/h-vetinari/cupy-feedstock/commit/ae6478432bedcba2c7b8b03af9399eff05938f9a), because that keeps all the cruft in the place we care about least, i.e. CUDA 11.8.